src/core/ext/client_config/client_channel.c

```diff
@@ -577,11 +607,15 @@ static void continue_picking(grpc_exec_ctx *exec_ctx, void *arg,
     /* cancelled, do nothing */
   } else if (error != GRPC_ERROR_NONE) {
     grpc_exec_ctx_sched(exec_ctx, cpa->on_ready, GRPC_ERROR_REF(error), NULL);
-  } else if (pick_subchannel(exec_ctx, cpa->elem, cpa->initial_metadata,
-                             cpa->initial_metadata_flags,
-                             cpa->connected_subchannel, cpa->on_ready,
-                             GRPC_ERROR_NONE)) {
-    grpc_exec_ctx_sched(exec_ctx, cpa->on_ready, GRPC_ERROR_NONE, NULL);
+  } else {
+    call_data *calld = cpa->elem->call_data;
+    gpr_mu_lock(&calld->mu);
```

Comment:
There is not actually any guarantee that some other thread won't be holding this lock.  If it is, then this thread will block until the other thread releases the lock.

I don't think this is any different from any other place where we do locking, so I'm not sure why this warrants a comment.  But perhaps I'm misunderstanding your question somehow?